pytradfri/device/light.py

```diff
@@ -27,69 +48,74 @@ class Light:
     pdf
     """
 
-    def __init__(self, device, index):
+    def __init__(self, device: Device, index: int):
         """Create object of class."""
         self.device = device
         self.index = index
 
     @property
-    def supported_features(self):
+    def supported_features(self) -> int:
         """Return supported features."""
-        return supported_features(self.raw)
+        return supported_features(self.raw.dict())
 
     @property
-    def state(self):
+    def state(self) -> bool:
         """Return device state."""
-        return self.raw.get(ATTR_DEVICE_STATE) == 1
+        return self.raw.state == 1
 
     @property
-    def dimmer(self):
+    def dimmer(self) -> int | None:
         """Return dimmer if present."""
         if self.supported_features & SUPPORT_BRIGHTNESS:
-            return self.raw.get(ATTR_LIGHT_DIMMER)
+            return self.raw.dimmer
         return None
 
     @property
-    def color_temp(self):
+    def color_temp(self) -> int | None:
         """Return color temperature."""
-        if self.supported_features & SUPPORT_COLOR_TEMP:
-            if self.raw.get(ATTR_LIGHT_MIREDS) != 0:
-                return self.raw.get(ATTR_LIGHT_MIREDS)
+        if self.supported_features & SUPPORT_COLOR_TEMP and self.raw.color_mireds:
+            return self.raw.color_mireds
         return None
 
     @property
-    def hex_color(self):
+    def hex_color(self) -> str | None:
         """Return hex color."""
         if self.supported_features & SUPPORT_HEX_COLOR:
-            return self.raw.get(ATTR_LIGHT_COLOR_HEX)
+            return self.raw.color_hex
         return None
 
     @property
-    def xy_color(self):
+    def xy_color(self) -> tuple[int, int] | None:
         """Return xy color."""
-        if self.supported_features & SUPPORT_XY_COLOR:
-            return (self.raw.get(ATTR_LIGHT_COLOR_X), self.raw.get(ATTR_LIGHT_COLOR_Y))
+        if (
+            self.supported_features & SUPPORT_XY_COLOR
+            and self.raw.color_xy_x is not None
+            and self.raw.color_xy_y is not None
+        ):
+            return (self.raw.color_xy_x, self.raw.color_xy_y)
         return None
 
     @property
-    def hsb_xy_color(self):
+    def hsb_xy_color(
```

Comment:
Maybe add checks for `None` here as well?

Reply:
https://github.com/home-assistant/core/blob/1ee4f47f8be3cd56f00e8fec35ce445436c80b82/homeassistant/components/tradfri/light.py#L200

Reply:
Side note for the future: Since we return xy color from another property I would expect this one to only return hsb color. Maybe add a property that does that and deprecate this one.

I agree that we should only return a tuple if all items are not None.

Reply:
Want me to deprecate here in this PR?

Reply:
Nah, just some thoughts for the future.